src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala

```diff
@@ -1086,8 +1087,42 @@ trait SilverTransforms extends SparkSessionWrapper {
         'organization_id, 'cluster_id, 'isRunning,
         'timestamp, 'state, 'current_num_workers, 'target_num_workers
       )
+      .withColumn("unixTimeMS_state_start", 'timestamp)
+      .withColumn("unixTimeMS_state_end", coalesce( // if state end open, use pipelineSnapTime, will be merged when state end is received
+        lead('timestamp, 1).over(stateUnboundW) - lit(1), // subtract 1 millis
+        lit(untilTime.asUnixTimeMilli)
+      ))
+
+    // Change for PR -934
+    // Get the ClusterID that has been Permenantly_Deleted
+    val clusterBaseDF = clusterBase(auditLogDF)
+    val clustersRemoved = clusterBaseDF
+      .filter('actionName.isin("permanentDelete"))
+      .select('timestamp.alias("deletion_timestamp"),'organization_id, 'cluster_id, 'userEmail.alias("deleted_by"))
+
+    val removedClusterID = clustersRemoved.select("cluster_id","deletion_timestamp").distinct()
+    val clusterEventsBaselineForRemovedCluster = clusterEventsBaseline.join(removedClusterID,Seq("cluster_id"))
 
-    clusterEventsBaseline
+    val window = Window.partitionBy('organization_id, 'cluster_id).orderBy('timestamp.desc)
+    val stateBeforeRemoval = clusterEventsBaselineForRemovedCluster.withColumn("rnk",rank().over(window)).filter('rnk === 1).drop("rnk")
+
+    val stateDuringRemoval = stateBeforeRemoval
+      .withColumn("timestamp",col("deletion_timestamp"))
+      .withColumn("isRunning",lit(false))
+      .withColumn("state",lit("PERMENANT_DELETE"))
+      .withColumn("current_num_workers",lit(0))
+      .withColumn("target_num_workers",lit(0))
+      .withColumn("unixTimeMS_state_start",'timestamp)
+      .withColumn("unixTimeMS_state_end",'timeStamp)
+      .drop("deletion_timestamp")
+    val columns: Array[String] = clusterEventsBaseline.columns
+
+    val stateDuringRemovalFinal = stateBeforeRemoval.withColumn("unixTimeMS_state_end",'deletion_timestamp).drop("deletion_timestamp").union(stateDuringRemoval).select(columns.map(col): _*)
```

Comment:
`unixTimeMS_state_end` should be `deletion_timestamp - 1` have the previous state end at the milisecond before the delete that way it follows the existing convention. Goal is illustrated in image attached

![image](https://github.com/databrickslabs/overwatch/assets/10840635/2d68c631-d30f-447d-8e00-2b70d9c84517)

Reply:
Done